addons/sale/models/sale.py

```diff
@@ -974,9 +974,9 @@ def _get_share_url(self, redirect=False, signup_partner=False, pid=None):
             return self.get_portal_url(query_string='&%s' % auth_param)
         return super(SaleOrder, self)._get_share_url(redirect, signup_partner, pid)
 
-    def _get_payment_type(self):
+    def _get_payment_type(self, default_type=False):
         self.ensure_one()
-        return 'form'
+        return default_type or 'form'
```

Comment:
This looks weird to me because it is the customer, the controller, and the acquirer that together determine whether the transaction must be tokenized. The way you propose to fix tokenization seems only logical given the way it is currently implemented but we could take the opportunity to make it a bit clearer without too much trouble:

```suggestion
    def _get_payment_type(self, tokenize=False):
        self.ensure_one()
        return 'form_save' if tokenize else 'form'
```

This way the method can simply be overridden in sale_subscription to 1. check if the order contains a sub product -> `contains_subscription_product` 2. return `super()._get_payment_type(tokenize=contains_subscription_product or tokenize)` which removes *a little bit* of confusion in this flow IMO.